src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1837,21 +1838,30 @@ private IDeclarationPatternOperation CreateBoundDeclarationPatternOperation(Boun
                 variable = ((BoundDiscardExpression)boundDeclarationPattern.VariableAccess).ExpressionSymbol;
             }
 
+            ITypeSymbol inputType = boundDeclarationPattern.InputType;
+            bool acceptsNull = boundDeclarationPattern.IsVar;
```

Comment:
>acceptsNull [](start = 17, length = 11)

What about `x is null`? If the constant is `null`, then we should probably set `AcceptsNull` to true.

Reply:
This is for the declaration pattern only.
The spec for a `var` pattern is that it accepts null.  The fact that it will never get null in some circumstances doesn't affect that fact.

---
In reply to: [244032712](https://github.com/dotnet/roslyn/pull/31967#discussion_r244032712) [](ancestors = 244032712)